### PR TITLE
sdk refactor wallet detection & support okx

### PR DESCRIPTION
**Pull request by wow-sven**
## Summary

- support oks
- refactor wallet detection process

- Closes #issue

### CODE REVIEW

**Comment by jolestar** [comment]
@yubing744 help review

File: sdk/typescript/rooch-sdk-kit/src/wellet/wallets.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+   * Add an event listener and subscribe to events for Wallets that are
+   * {@link WalletsEventsListeners.register | registered} and
+   * {@link WalletsEventsListeners.unregister | unregistered}.
+   *
+   * @param event    Event type to listen for. {@link WalletsEventsListeners.register | `register`} and
+   * {@link WalletsEventsListeners.unregister | `unregister`} are the only event types.
+   * @param listener Function that will be called when an event of the type is emitted.
+   *
+   * @return
+   * `off` function which may be called to remove the event listener and unsubscribe from events.
```
[the remark this answers is not in the document]
Comment (wow-sven) [reply sdk/typescript/rooch-sdk-kit/src/wellet/wallets.ts:35]: Annotations and implementations are consistent

File: sdk/typescript/rooch-sdk-kit/src/wellet/wallets.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+import { Wallet } from './wallet.js'
+
+let wallets: Wallets | undefined = undefined
+const registered = new Set<Wallet>()
```
[the remark this answers is not in the document]
Comment (wow-sven) [reply sdk/typescript/rooch-sdk-kit/src/wellet/wallets.ts:7]: The real environment should be the singleton pattern，
If we need parallelism to speed up our tests later, we can modify the mock to generate a unique one for each test

File: sdk/typescript/rooch-sdk-kit/src/wellet/wallets.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (yubing744) [reply sdk/typescript/rooch-sdk-kit/src/wellet/wallets.ts:35]: OK, I misread it.

File: sdk/typescript/rooch-sdk-kit/src/wellet/wallets.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+  /**
+   * Register Wallets. This can be used to programmatically wrap non-standard wallets as Standard Wallets.
+   *
+   * Apps generally do not need to, and should not, call this.
+   *
+   * @param wallets Wallets to register.
+   *
+   * @return
+   * `unregister` function which may be called to programmatically unregister the registered Wallets.
```
[the remark this answers is not in the document]
Comment (yubing744) [reply sdk/typescript/rooch-sdk-kit/src/wellet/wallets.ts:49]: OK, I misread it.

File: sdk/typescript/rooch-sdk-kit/src/wellet/wallets.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (yubing744) [reply sdk/typescript/rooch-sdk-kit/src/wellet/wallets.ts:7]: If it is not convenient to define it as a class here, this is also a solution.

**Review by yubing744 — APPROVED** [review]

**Pull request merged**